lib/redisOpenAM.js

```diff
@@ -40,39 +36,38 @@ function createBasicStrategy(options) {
                     password: password,
                     scope: scopes
                 },
-                error: true
+                error: true,
+                json:true
             });
         }
 
-        function cacheAndGetTokenInfo(res, body) {
-            var parsedBody = JSON.parse(body);
-
-            foundToken = parsedBody.access_token;
-
-            return promise.all([
-                cacheToken(parsedBody),
-                getTokenInfo()
-            ]);
+        function getandCacheTokenInfo(res, body) {
+            return getTokenInfo(body)
+                .spread(cacheToken);
         }
 
-        function cacheToken(parsedBody) {
-            var expiry = parsedBody.expires_in;
+        function getTokenInfo(tokenBody) {
+            return $http.get(infoURL+'?access_token='+tokenBody.access_token, {json: true})
+                .spread(function(res, infoBody) {
+                    if (infoBody.error ||
```

Comment:
isn't the default http-as-promised  behaviour to throw an error in case the response status is not reporting success ?
my  guess is that if you want to check the body you should be setting the error: false option
don't think this behaviour is covered with a test, is it?